src/librustc_driver/lib.rs

```diff
@@ -1224,9 +1224,20 @@ pub fn install_ice_hook() {
 }
 
 /// This allows tools to enable rust logging without having to magically match rustc's
-/// log crate version
+/// log crate version.
 pub fn init_rustc_env_logger() {
-    env_logger::init_from_env("RUSTC_LOG");
+    init_env_logger("RUSTC_LOG")
+}
+
+/// This allows tools to enable rust logging without having to magically match rustc's
+/// log crate version. In contrast to `init_rustc_env_logger` it allows you to choose an env var
+/// other than `RUSTC_LOG`.
+pub fn init_env_logger(env: &str) {
+    let builder = tracing_subscriber::FmtSubscriber::builder();
+
+    let builder = builder.with_env_filter(tracing_subscriber::EnvFilter::from_env(env));
+
+    builder.init()
 }
```

Comment:
Some of these methods are deprecated. Consider writing the `tracing_subscriber::fmt::Subscriber`-specific of:

```rust
tracing_subscriber::fmt()
    .with_env_filter(tracing_subscriber::EnvFilter::from_env(env))
    .init();
```

Reply:
None of the methods used here are deprecated?

Reply:
Whoops! I got them mixed up with [`fmt::Layer::builder`](https://docs.rs/tracing-subscriber/0.2.9/tracing_subscriber/fmt/struct.Layer.html#method.builder) and [`fmt::LayerBuilder::finish`](https://docs.rs/tracing-subscriber/0.2.9/tracing_subscriber/fmt/type.LayerBuilder.html#method.finish).